src/main/java/com/app/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.app.onlinebookstore.repository.impl;
+
+import com.app.onlinebookstore.model.Book;
+import com.app.onlinebookstore.repository.BookRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert book " + book, e);
```

Comment:
it is better to create your own error, I have seen this correction from mentors among colleagues